app/models/calculator/weight.rb

```diff
@@ -57,8 +57,13 @@ def quantity_implied_in_final_weight_volume(line_item)
       (1.0 * line_item.final_weight_volume / line_item.variant.unit_value).round(3)
     end
 
+    def use_final_weight_volume?(line_item)
+      final_weight_volume_present?(line_item) && !line_item.variant.unit_value.zero?
+    end
+
     def final_weight_volume_present?(line_item)
-      line_item.respond_to?(:final_weight_volume) && line_item.final_weight_volume.present?
+      line_item.respond_to?(:final_weight_volume) && line_item.final_weight_volume.present? &&
+        !line_item.final_weight_volume.zero?
```

Comment:
looking at the DB I suspect a zero final_weight_volume is actually meaningful (set by user) and we should use it... I'd not change that behavior.